blockchain/blockdao/blockdao.go

```diff
@@ -719,9 +743,15 @@ func (dao *blockDAO) putBlock(blk *block.Block) error {
 	}
 	batchForBlock := batch.NewBatch()
 	hash := blk.HashBlock()
+	heightKey := heightKey(blkHeight)
 	batchForBlock.Put(blockHeaderNS, hash[:], serHeader, "failed to put block header")
 	batchForBlock.Put(blockBodyNS, hash[:], serBody, "failed to put block body")
 	batchForBlock.Put(blockFooterNS, hash[:], serFooter, "failed to put block footer")
+	if dao.ContainsSystemLog() {
```

Comment:
if the DB file supports system log
1. generate all systemlog from a block's receipt
2. store if systemlog exists

Reply:
is this for old db?if old db is not sync from height 1,so they will not have systemlog forever?

Reply:
that's right
the plan is after this PR merge, Yutong will restart to sync from block 1